core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -59,6 +76,10 @@ protected TableScanContext context() {
     return context;
   }
 
+  protected ScanReporter scanReporter() {
```

Comment:
Does this need to be exposed publicly?

Reply:
subclasses need access to this and checkstyle is complaining that this needs to be exposed via a method for subclasses

Reply:
moved the `ScanReporter` into the `TableScanContext`, so this change isn't required anymore